File: src/contests/cg/puzzle/the-descent.ts
// @ts-nocheck
// 🎮 CodinGame Puzzle - the-descent
// https://www.codingame.com/training/easy/the-descent

// Game loop: each turn, read 8 mountain heights and fire on the tallest one.
while (true) {
    let maxHeight = -1;
    let maxIndex = 0;

    for (let i = 0; i < 8; i++) {
        const mountainH = parseInt(readline());
        if (mountainH > maxHeight) {
            maxHeight = mountainH;
            maxIndex = i;
        }
    }

    console.log(maxIndex);
}

File: src/contests/cg/puzzle/python3/the-descent.py
# 🎮 CodinGame Puzzle - the-descent
# https://www.codingame.com/training/easy/the-descent

# Game loop: each turn, read 8 mountain heights and fire on the tallest one.
while True:
    max_height = -1
    max_index = 0
    for i in range(8):
        mountain_h = int(input())
        if mountain_h > max_height:
            max_height = mountain_h
            max_index = i
    print(max_index)
